### DISCUSSION TITLE
I am trying to use Browser mode to run some components, but I need to set some libraries variables using script tags. Is that possible?

### DISCUSSION THREAD

**Discussion by matheus-hatzlhoffer** [vitest-dev/vitest#6467] [Q&A] [answered]
Hello everyone. I am trying to test some components using Web Browser. I tried to test it using JSDOM, but because it is not a real environment doesn't have all the capabilities to describe the test. Like described here #5828. 

I tried to run in browser mode, but to configure the environment I need to run some script tags inside the HTML dom, because this third party library doesn't follow ES6 modularization. In JSDOM I instantiate it using `{runScripts: "dangerously", resources: "usable"}` configuration and use this code snippet to run the script tag:

```
// Adding ISC_Core.js script
    const iscCoreContent = fs.readFileSync(
      path.resolve("./isomorphic/system/modules/", "ISC_Core.js"),
      "utf8"
    );
    const iscCoreElement = document.createElement("script");
    iscCoreElement.textContent = iscCoreContent;
    document.head.appendChild(iscCoreElement);
```

But in Browser Mode I couldn't find any resources to run HTML scripts tags or to import using some code snippet like the above. If I try the same approach I receive `Error: Module "fs" has been externalized for browser compatibility. Cannot access "fs.readFileSync" in client code.` 

How can I run script tags in Browser Mode?

**Comment by sheremet-va** [comment, marked as answer]
You can run files using [commands](https://vitest.dev/guide/browser/commands.html#files-handling) or inject scripts by defining [`testerScripts`](https://vitest.dev/config/#browser-testerscripts) in the config.

**Comment by matheus-hatzlhoffer** [comment]
With the second approach the scripts ran with some error `TypeError: Right-hand side of 'instanceof' is not callable` , do you know what can it be?

-------------

I tried to inject using [testerScripts](https://vitest.dev/config/#browser-testerscripts) but it didn't work. It is not clear to me if the array should have the path to the script. I tried: 
```
testerScripts: [
          "./isomorphic/system/modules/ISC_Core.js",
          "./isomorphic/system/modules/ISC_Foundation.js",
          "./isomorphic/system/modules/ISC_Containers.js",
          "./isomorphic/system/modules/ISC_Grids.js",
          "./isomorphic/system/modules/ISC_Forms.js",
          "./isomorphic/system/modules/ISC_DataBinding.js",
        ],
```

but I received the error `ReferenceError: isc is not defined` so it didn't run

**Reply by sheremet-va** [reply]
`testerScripts` is not an array of paths, it's a an array of `BrowserScript` objects:

```ts
{
  /**
   * If "content" is provided and type is "module", this will be its identifier.
   *
   * If you are using TypeScript, you can add `.ts` extension here for example.
   * @default `injected-${index}.js`
   */
  id?: string
  /**
   * JavaScript content to be injected. This string is processed by Vite plugins if type is "module".
   *
   * You can use `id` to give Vite a hint about the file extension.
   */
  content?: string
  /**
   * Path to the script. This value is resolved by Vite so it can be a node module or a file path.
   */
  src?: string
  /**
   * If the script should be loaded asynchronously.
   */
  async?: boolean
  /**
   * Script type.
   * @default 'module'
   */
  type?: string
}
```

If you use `defineConfig` helper from `vitest/config`, you can see the JSDoc helper for those.

**Comment by matheus-hatzlhoffer** [comment]
I was able to get it using commands
